Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -927,4 +927,120 @@ namespace MiscCmds
 		cmds->Print(L"OK\n");
 		return;
 	}
+
+	void SetPlayerHp(uint clientId, float percentage)
+	{
+		Players[clientId].fRelativeHealth = percentage;
+		HookClient->Send_FLPACKET_SERVER_SETHULLSTATUS(clientId, percentage);
+	}
+
+	void SetPlayerFuse(uint clientId, uint fuseId)
+	{
+		IObjInspectImpl *obj = HkGetInspect(clientId);
+		if (obj)
+		{
+			HkLightFuse((IObjRW*)obj, fuseId, 0.0f, 0.0f, 0.0f);
+		}
+	}
+
+	void UnsetPlayerFuse(uint clientId, uint fuseId)
+	{
+		IObjInspectImpl *obj = HkGetInspect(clientId);
+		if (obj)
+		{
+			HkUnLightFuse((IObjRW*)obj, fuseId, 0.0f);
+		}
+	}
+
+	uint CheckAdminRightAndGetTargetClient(CCmds* cmds, const wstring& charName)
+	{
+		if (!(cmds->rights & RIGHT_EVENTMODE))
+		{
+			cmds->Print(L"ERR No permission\n");
+			return 0;
+		}
+		uint targetClient = 0;
+		if (charName.empty())
+		{
+			wstring adminName = cmds->GetAdminName();
+			if (adminName.empty())
+			{
+				cmds->Print(L"ERR No shipname provided\n");
+				return 0;
+			}
+			uint clientId = HkGetClientIdFromCharname(adminName);
+			uint shipId;
+			uint targetId;
+			pub::Player::GetShip(clientId, shipId);
```

Comment:
isn't this going to fall apart if the admin is console/socket? or the admin is an in-game player but does not have a player selected?

Reply:
Those scenarios will cause an invalid targetClient value to be produced on either line 976 or 980, that will then be caught on line 985.